bin.src/create_agn_db.py

```diff
@@ -208,30 +208,100 @@ def get_m_i(abs_mag_i, redshift):
-                      + '"agn_sfz": %.3e, "agn_sfy": %.3e}}' % (sfz, sfy))
-                 for ii, mm, ss, tt, sfu, sfg, sfr, sfi, sfz, sfy in
-                 zip(galaxy_id, mag_norm, seed_arr, tau,
-                     sf_dict['u'], sf_dict['g'], sf_dict['r'], sf_dict['i'],
-                     sf_dict['z'], sf_dict['y']))
-
-        cursor.executemany('INSERT INTO agn_params VALUES(?, ?, ?)', vals)
+        chunk_size = 100000
+        full_size = len(redshift_full)
+        print('starting iteration')
+        t_start = time.time()
+        ct_simulated = 0
+        for i_start in range(0, full_size, chunk_size):
+            i_end = i_start + chunk_size
+            duration = (time.time()-t_start)/3600.0
+            per = duration/(1+i_start)
+            predicted = full_size*per
+            print("    %d through %d -- took %.2e hrs; predict %.2e" %
+            (i_start,i_end,duration,predicted))
+
+            galaxy_id = galaxy_id_full[i_start:i_end]
+            ra = ra_full[i_start:i_end]
+            dec = dec_full[i_start:i_end]
+            ct_simulated += len(galaxy_id)
+            redshift = redshift_full[i_start:i_end]
+            log_edd_ratio = log_edd_ratio_full[i_start:i_end]
+            bhm = bhm_full[i_start:i_end]
+
+            abs_mag_i = M_i_from_L_Mass(log_edd_ratio, np.log10(bhm))
+
+            obs_mag_i = get_m_i(abs_mag_i, redshift)
+
+            if args.m_i_cut is not None:
+                valid = np.where(obs_mag_i <= args.m_i_cut)
+                redshift = redshift[valid]
+                bhm = bhm[valid]
+                log_edd_ratio = log_edd_ratio[valid]
+                abs_mag_i = abs_mag_i[valid]
+                galaxy_id = galaxy_id[valid]
+                ra = ra[valid]
+                dec = dec[valid]
+                obs_mag_i = obs_mag_i[valid]
+
+            tau = tau_from_params(redshift,
+                                  abs_mag_i,
+                                  bhm, rng=rng)
+
+            sf_dict = {}
+            for bp in ('u', 'g', 'r', 'i', 'z', 'y'):
+                eff_wavelen = 10.0*bp_dict[bp].calcEffWavelen()[0]
+                sf_dict[bp] = SF_from_params(redshift, abs_mag_i,
+                                             bhm, eff_wavelen,
+                                             rng=rng)
+
+            # cut on structure function value
+            for bp in 'ugrizy':
+                valid = np.where(sf_dict[bp]<args.max_sf)
+                redshift = redshift[valid]
+                tau = tau[valid]
+                log_edd_ratio = log_edd_ratio[valid]
+                abs_mag_i = abs_mag_i[valid]
+                obs_mag_i = obs_mag_i[valid]
+                bhm = bhm[valid]
+                galaxy_id = galaxy_id[valid]
+                ra = ra[valid]
+                dec = dec[valid]
+                for other_bp in 'ugrizy':
+                    sf_dict[other_bp] = sf_dict[other_bp][valid]
```

Comment:
I don't understand the logic of Lines 278-279, but you can ignore me if that should be obvious to others..

Reply:
I am trying to make sure that only AGN who have structure function values in all bands less than the allowable maximum are kept.  I have added comments to explain this.